avocado/job.py

```diff
@@ -109,6 +109,22 @@ def __init__(self, args=None):
         self.status = "RUNNING"
         self.result_proxy = result.TestResultProxy()
         self.sysinfo = None
+        self._setup_timeout()
+
+    def _setup_timeout(self):
+        units = {'s': 1, 'm': 60, 'h': 60 * 60, 'd': 60 * 60 * 24}
+        mult = 1
+        timeout = getattr(self.args, 'job_timeout', None)
+        if timeout is not None:
+            if timeout[-1].lower() in units.keys():
+                mult = units[timeout[-1].lower()]
+                timeout = timeout[:-1]
+            try:
+                self.timeout = int(timeout) * mult
+            except ValueError:
+                self.timeout = 0
+        else:
+            self.timeout = 0
```

Comment:
I'd keep the parsing of the units at the command line parsing. That way, you can fail early if the value is invalid, and give the job a simple value in seconds and nothing to worry about! :)